api/api.py

```diff
@@ -24,23 +23,24 @@ def get_co2_estimate(transport_mode=None, distance_km=None, vehicle_occupancy=No
         # Calculate distance from origin/destination
         origin = (origin_lat, origin_lon)
         destination = (destination_lat, destination_lon)
-
         distance_km = haversine(origin, destination)
     else:
-        return "Not enough information was provided to calculate CO2 estimate."
+        return {
+            "error": "Not enough information was provided to calculate CO2 estimate."
+        }
 
-    co2_estimate = estimate_co2(
-        mode=transport_mode, distance_in_km=distance_km, occupancy=vehicle_occupancy)
+    co2_estimate = estimate_co2(mode=transport_mode, distance_in_km=distance_km, occupancy=vehicle_occupancy)
 
     return_data = {
-        "transport_mode": transport_mode,
+        "transport_mode": str(transport_mode),
```

Comment:
While we are working here, can we also return the `distance_km` ?

Reply:
Also, `vehicle_occupancy` is return as `None` if not explicitly set, but should be the value actually used, i.e., default from `Mode` if not set.

Reply:
@brylie  Good call on `distance_km` because we calculate that 👍 

@MarkusShepherd Regarding default occupancy, how to do it properly? 🤔  `estimate_co2` defaults internally to average_occupancy so either we should do it ourselves before calling it or return value from `estimate_co2`

Reply:
I'd say we need to explicitly find the transport mode and use its default occupancy:

```python
from transport_co2 import Mode
if not isinstance(transport_mode, Mode):
    try:
        transport_mode = Mode[str(transport_mode).upper()]
    except Exception:
        LOGGER.exeception('unknown mode %r', transport_mode)
       return {'error': f'unknown mode {transport_mode}'}
    vehicle_occupancy = vehicle_occupancy or transport_mode.average_occupancy
```

Reply:
Then we can of course use this in the end:

```python
    return_data = {
        "transport_mode": transport_mode.name,
        "vehicle_occupancy": vehicle_occupancy,
        "co2_estimate": co2_estimate,
       "distance_in_km": distance_km,
    }
```

Reply:
fixed by using `transport_mode.name if isinstance(transport_mode, Mode) else transport_mode`